src/main/java/org/folio/marc4ld/service/marc2ld/mapper/mapper/OtherVersionMapper.java

```diff
@@ -0,0 +1,183 @@
+import static org.folio.marc4ld.util.Constants.A;
+import static org.folio.marc4ld.util.Constants.B;
+import static org.folio.marc4ld.util.Constants.SUBFIELD_SRS_ID;
+import static org.folio.marc4ld.util.Constants.T;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.folio.ld.dictionary.PredicateDictionary;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+import org.folio.ld.dictionary.model.Resource;
+import org.folio.ld.dictionary.model.ResourceEdge;
+import org.folio.ld.fingerprint.service.FingerprintHashService;
+import org.folio.marc4ld.dto.MarcData;
+import org.folio.marc4ld.service.label.LabelService;
+import org.folio.marc4ld.service.marc2ld.mapper.Marc2ldMapper;
+import org.marc4j.marc.DataField;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class OtherVersionMapper implements Marc2ldMapper {
+
+  private static final String TAG = "776";
+  private static final Set<ResourceTypeDictionary> LOCAL_ID_TYPES = Set.of(IDENTIFIER, ID_LOCAL);
+  private static final Set<ResourceTypeDictionary> IDENTIFIER_TYPE = Set.of(IDENTIFIER);
+  private static final String DLC_PREFIX = "DLC";
+
+  private final MapperHelper mapperHelper;
+  private final LabelService labelService;
+  private final FingerprintHashService hashService;
+
+  @Override
+  public String getTag() {
+    return TAG;
+  }
+
+  @Override
+  public boolean canMap(PredicateDictionary predicate) {
+    return predicate == OTHER_VERSION;
+  }
+
+  @Override
+  public void map(MarcData marcData, Resource resource) {
+    var workAndInstanceTitles = getWorkAndInstanceTitles(marcData);
+    var properties = mapperHelper.getProperties(resource);
+    properties.put(LABEL.getValue(), List.of(workAndInstanceTitles.workTitle()));
+    resource.setDoc(mapperHelper.getJsonNode(properties));
+    mapTitle(resource, workAndInstanceTitles.workTitle());
+    mapInstance(resource, workAndInstanceTitles.instanceTitle());
+    adjustWorkToInstanceRelation(resource);
+    resource.setLabel(workAndInstanceTitles.workTitle());
+  }
+
+  private WorkAndInstanceTitles getWorkAndInstanceTitles(MarcData marcData) {
+    String workTitle;
+    String instanceTitle;
+    var dataField = marcData.getDataField();
+    if (isSubfieldPresent(SUBFIELD_SRS_ID, dataField)) {
+      workTitle = getSubfieldValue(SUBFIELD_SRS_ID, dataField);
+      if (isSubfieldPresent(T, dataField)) {
+        instanceTitle = getSubfieldValue(T, dataField);
+      } else {
+        instanceTitle = getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField);
+      }
+    } else if (isSubfieldPresent(T, dataField)) {
+      return new WorkAndInstanceTitles(getSubfieldValue(T, dataField), getSubfieldValue(T, dataField));
+    } else {
+      return new WorkAndInstanceTitles(
+        getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField),
```

Comment:
This is not correct. We have to get the title value for 245$a and 245$b. I know it is not straight forward as we do not have MARC 245 available in this method. Let us think how to do that.

Reply:
Done